### ISSUE TITLE
Multiple databroker urls returns x509 validation error

### ISSUE THREAD

**Issue by ventris** [pomerium/pomerium#2429]
## What happened?

Pomerium seems to incorrectly validate certificates for databroker_service_urls when you add multiple items to the list.

## What did you expect to happen?

Pomerium to be able to validate each FQDN separately when it does its connection.

## How'd it happen?

1. Use `databroker_service_urls` with multiple hosts
2. Start Pomerium with `log_level: debug` and `debug: true`

## What's your environment like?

- Pomerium version (retrieve with `pomerium --version`):
```
{"level":"info","version":"0.14.7-1624903390+e5d4c824","time":"2021-08-04T07:30:17Z","message":"cmd/pomerium"}
```
- Server Operating System/Architecture/Cloud:
Docker container running on Ubuntu 18.04

## What's your config.yaml?

```config.yaml
address: :443

databroker_service_urls:
        - "https://databroker1.example.com:5001"
        - "https://databroker2.example.com:5001"
        - "https://databroker3.example.com:5001"

databroker_storage_tls_skip_verify: true

authorize_service_url: "https://authorize.example.com"

authenticate_service_url: "https://authenticate.example.com"

shared_secret: "redacted"
cookie_secret: "redacted"
```

## What did you see in the logs?

```logs
{"level":"error","error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate is valid for databroker1.example.com, 10.10.10.10, 127.0.0.1, localhost, not databroker1.example.com:5001,databroker2.example.com:5001,databroker3.example.com:5001\"","time":"2021-08-02T09:02:41Z","message":"controlplane: error storing configuration event"}
```

## Additional context

If I add all hostnames to the same certificate I still get this error.

From the looks of the error log it seems like Pomerium is concatenating all items in the list before trying to validate the certificate when it does the connection.

If I use `databroker_service_urls` with just one item it works fine but when I have more than 1 item I get the above error every time. Same goes when trying to use a wildcard certificate.

@travisgroth

**Comment by calebdoxsey** [comment]
The issue here is caused by our custom resolver. It supports URLs like:

```
pomerium:///databroker1.example.com,databroker2.example.com
```

Although it establishes the connections properly, the TLS verification is done using the server name:

```
databroker1.example.com,databroker2.example.com
```

And obviously that doesn't work. We could easily override the server name, but unfortunately it's possible each databroker server will have independent TLS certificates, so we can't rely on a single name.

One solution might be to turn on `InsecureSkipVerify` and then define our own `VerifyConnection` function that is more lenient about naming. (I'm wary about doing this because it would be easy to miss something subtle)

Its very strange that the resolving API doesn't have some mechanism for interfacing better with TLS, but unfortunately I couldn't find anything that would work at that level.

A second solution, and probably the more preferable one, would be to send gRPC traffic through the local envoy server and let it handle TLS verification and load balancing. This should fix the problem but will take more time to implement.

**Comment by calebdoxsey** [comment]
We now send client traffic through envoy, which should fix this issue in v0.15.
